layers/+source-control/git/README.org

```diff
@@ -59,6 +65,163 @@ file.
+
+- [[https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account][GitHub documentation]]
+- [[https://docs.gitlab.com/ee/ssh/#add-an-ssh-key-to-your-gitlab-account][GitLab documentation]]
+ 
+**** Making Emacs and Magit SSH login–keyring–aware 
+After following forge-specific documentation (linked to in the above heading)
+for using SSH keys with that forge, set the variable ~auth-sources~ to include
+the string(s) ="secrets:Login"= (to use a login keyring, which will be unlocked by
+your desktop environment upon login), and/or ="secrets:session"= (note the
+lowercase =s= in =session=; the session option will "open [the keyring] only for the
+lifetime of Emacs", as the /GNU Emacs Manual/ states [Free Software Foundation
+2021]).
+
+1. Configure Emacs to use login or session keyrings. The default configuration of
+   Emacs (and Spacemacs) does not include this. Login and session keyrings are
+   managed by the ssh-agent of the desktop environment you use on Linux.
+
+   - If you use Windows, you will need to consult extra documentation not
+     discussed here.
+
+   Add the following to your ~spacemacs/user-config~ function, as shown below.
+
+   #+begin_src emacs-lisp
+     (defun spacemacs/user-config ()
+     "Configuration for user code: This function is called at the very end of
+     Spacemacs startup, after layer configuration. Put your configuration code here,
+     except for variables that should be set
+     before packages are loaded."
+
+       (setq auth-sources '(default
+                             "secrets:Login")))
+   #+end_src
+
+2. Add the key to the SSH configuration file, so that when an SSH connection is
+   made to a Git forge, the appropriate key is used with that host by the
+   ssh-agent.
+
+   #+begin_src emacs-lisp
+     ((lambda (forgeUsername
+               sshKeyName
+               (&optional (gitForgeProperName "Github")
+                          (gitForgeHostname   "github.com")
+                          (addKeyToAgent-p    "yes")))
+        "Add a new SSH key to the user's login keyring, and\
+             configure it for a Git forge."
+        nil
+        (append-to-file
+         (concat "\n"
+                 "HOST " gitForgeProperName "\n"
+
+                 ;; The .pub file extension is not necessary; it may cause
+                 ;; issues to include it. See man 5 ssh_config
+                 "     IdentityFile ~/.ssh/" sshKeyName "\n"
+                 "     User " forgeUsername "\n"
+                 "     Hostname " gitForgeHostname "\n"
+                 "     AddKeysToAgent " addKeyToAgent-p)
+         nil
+         "~/.ssh/config"))
+
+      "Your Git forge username"
+      "The path to your SSH key"
+
+      ;; ;; Optionally specify alternative to GitHub, and not add key to agent
+      ;; ;; permanently.
+      ;; "GitLab"
+      ;; "gitlab.com"
+      ;; ;; Either "no", or "confirm".
+      ;; "no"
+      )
+   #+end_src
```

Comment:
It seems this script adds a host configuration to `~/.ssh/config`, which seems simpler to showing an example of that host configuration, e.g. for GitHub
```
HOST GitHub
IdentityFile ~/.ssh/rsa_id
User forgeusername
Hostname github.com
AddKeysToAgent yes
```

Reply:
Yes. Let's don't touch ~/.ssh/config and let user edit it manually.